lib/src/native/realm_core.dart

```diff
@@ -47,9 +47,26 @@ import '../user.dart';
 import '../set.dart';
 import 'realm_bindings.dart';
 
-late RealmLibrary _realmLib;
+const bugInTheSdkMessage = "This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues";
 
-final _RealmCore realmCore = _RealmCore();
+final _realmLib = () {
```

Comment:
Is this change related to decimal128?

Reply:
Yes. Allows `lib` to be used before `_RealmCore` is instantiated.

Reply:
I would suggest we move decimal128.dart into the native directory and keeping the initialization of realmCore as is. 
_realmLib is intentionally hidden from the rest of the SDK. We could declare deciam128 as a partial file to realm_core.dart if we want to keep it separate from the rest of the native invocations. Which I agree is better.